platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/validation/EventSignatureValidator.java

```diff
@@ -16,248 +16,37 @@
-                    "Cannot validate events for software version {} that is greater than the current software version {}",
-                    eventVersion,
-                    currentSoftwareVersion);
-            return null;
-        } else if (softwareComparison > 0) {
-            // current software version is greater than event software version
-            if (previousAddressBook == null) {
-                rateLimitedLogger.error(
-                        EXCEPTION.getMarker(),
-                        "Cannot validate events for software version {} that is less than the current software version {} without a previous address book",
-                        eventVersion,
-                        currentSoftwareVersion);
-                return null;
-            }
-            return previousAddressBook;
-        } else {
-            // current software version is equal to event software version
-            return currentAddressBook;
-        }
-    }
-
-    /**
-     * Determine whether a given event has a valid signature.
-     *
-     * @param event the event to be validated
-     * @return true if the event has a valid signature, otherwise false
-     */
-    private boolean isSignatureValid(@NonNull final GossipEvent event) {
-        final AddressBook applicableAddressBook = determineApplicableAddressBook(event);
-        if (applicableAddressBook == null) {
-            // this occurrence was already logged while attempting to determine the applicable address book
-            return false;
-        }
-
-        final NodeId eventCreatorId = event.getHashedData().getCreatorId();
-
-        if (!applicableAddressBook.contains(eventCreatorId)) {
-            rateLimitedLogger.error(
-                    EXCEPTION.getMarker(),
-                    "Node {} doesn't exist in applicable address book. Event: {}",
-                    eventCreatorId,
-                    event);
-            return false;
-        }
-
-        final PublicKey publicKey =
-                applicableAddressBook.getAddress(eventCreatorId).getSigPublicKey();
-        if (publicKey == null) {
-            rateLimitedLogger.error(
-                    EXCEPTION.getMarker(), "Cannot find publicKey for creator with ID: {}", eventCreatorId);
-            return false;
-        }
-
-        final boolean isSignatureValid = signatureVerifier.verifySignature(
-                event.getHashedData().getHash().getValue(),
-                event.getUnhashedData().getSignature(),
-                publicKey);
-
-        if (!isSignatureValid) {
-            rateLimitedLogger.error(
-                    EXCEPTION.getMarker(),
-                    "Event failed signature check. Event: {}, Signature: {}, Hash: {}",
-                    event,
-                    CommonUtils.hex(event.getUnhashedData().getSignature()),
-                    event.getHashedData().getHash());
-        }
-
-        return isSignatureValid;
-    }
+public interface EventSignatureValidator {
```

Comment:
Missing a doc for the interface.

The pattern I've been following is to transfer the doc from the original class into the interface, and then give the impl class a doc that just says it's the default implementation of the interface.

Reply:
thanks for the catch!
updated